tests/pow_complex.cpp

```diff
@@ -192,13 +221,13 @@ TEMPLATE_TEST_CASE("Test complex pow deci-cplx overload", "[pow]", double,
      }).wait();
     Q.copy(d_cplx_out, &h_cplx_out, 1).wait();
 
-    check_results(h_cplx_out, std_out);
+    CHECK_THAT(h_cplx_out, ComplexWithinULP(std_out, 1));
   }
 
   // Check cplx::complex output from host
   h_cplx_out = sycl::ext::cplx::pow<T>(deci_input, cplx_input);
 
-  check_results(h_cplx_out, std_out);
+  CHECK_THAT(h_cplx_out, ComplexWithinULP(std_out, 1));
```

Comment:
CHECK_THAT is from Catch2, but we could provide a custom macro that makes it more compact, if the macro nesting works out (I think it will). Something like `CHECK_COMPLEX_WITH_ULP(check_value, expected_value, tol_multiplier)`

Reply:
I think CHECK_COMPLEX_WITHIN_ULP is still worth it for readability / type-ability, the line number is there and someone encountering it can look up the macro definition and then it will make sense.